### PR TITLE
Taxon and taxon tree target types

### PR DESCRIPTION
**Pull request by RandyCupic**
This PR deprecates the existing `sylius_taxon` target type, which has been named as a single taxon one but actually behaves as a taxon tree target type, and changes it's label to `Taxon tree`.

It also introduces a new `sylius_single_taxon` target type, which is also already deprecated.

In the new major release, this should be cleaned up so that we have `sylius_taxon` and `sylius_taxon_tree` target types.

### CODE REVIEW

File: bundle/Resources/config/services/layout_resolver.yaml [hunk 1]
```diff
@@ -29,16 +29,31 @@ services:
         tags:
             - { name: netgen_layouts.target_type.doctrine_handler, target_type: sylius_taxon_product }
 
+    netgen_layouts.sylius.layout_resolver.target_type.single_taxon:
```
Comment (emodric) [inline bundle/Resources/config/services/layout_resolver.yaml:32]: Please keep these three definitions together, as is the case in the rest of the file.

File: lib/Layout/Resolver/Form/TargetType/Mapper/SingleTaxon.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+declare(strict_types=1);
+
+namespace Netgen\Layouts\Sylius\Layout\Resolver\Form\TargetType\Mapper;
+
+use Netgen\ContentBrowser\Form\Type\ContentBrowserType;
+use Netgen\Layouts\Layout\Resolver\Form\TargetType\Mapper;
+
+/**
+ * @deprecated this class will be renamed to simply Taxon in next major release
```
Comment (emodric) [inline lib/Layout/Resolver/Form/TargetType/Mapper/SingleTaxon.php]: Please use concrete version: `in 2.0`

File: lib/Layout/Resolver/Form/TargetType/Mapper/Taxon.php [hunk 3]
```diff
@@ -7,6 +7,9 @@
 use Netgen\ContentBrowser\Form\Type\ContentBrowserType;
 use Netgen\Layouts\Layout\Resolver\Form\TargetType\Mapper;
 
+/**
+ * @deprecated this class will be renamed to TaxonTree in next major release
```
Comment (emodric) [inline lib/Layout/Resolver/Form/TargetType/Mapper/Taxon.php]: Please use concrete version: `in 2.0`

File: lib/Layout/Resolver/TargetHandler/Doctrine/SingleTaxon.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+declare(strict_types=1);
+
+namespace Netgen\Layouts\Sylius\Layout\Resolver\TargetHandler\Doctrine;
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use Doctrine\DBAL\Types\Types;
+use Netgen\Layouts\Persistence\Doctrine\QueryHandler\TargetHandlerInterface;
+
+/**
+ * @deprecated this class will be renamed to simply Taxon in next major release
```
Comment (emodric) [inline lib/Layout/Resolver/TargetHandler/Doctrine/SingleTaxon.php]: Please use concrete version: `in 2.0`

File: lib/Layout/Resolver/TargetHandler/Doctrine/Taxon.php [hunk 5]
```diff
@@ -8,6 +8,9 @@
 use Doctrine\DBAL\Query\QueryBuilder;
 use Netgen\Layouts\Persistence\Doctrine\QueryHandler\TargetHandlerInterface;
 
+/**
+ * @deprecated this class will be renamed to TaxonTree in next major release
```
Comment (emodric) [inline lib/Layout/Resolver/TargetHandler/Doctrine/Taxon.php]: Please use concrete version: `in 2.0`

File: lib/Layout/Resolver/TargetType/SingleTaxon.php [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+namespace Netgen\Layouts\Sylius\Layout\Resolver\TargetType;
+
+use Netgen\Layouts\Layout\Resolver\TargetType;
+use Netgen\Layouts\Sylius\Validator\Constraint as SyliusConstraints;
+use Sylius\Component\Taxonomy\Model\TaxonInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Validator\Constraints;
+
+/**
+ * @deprecated this class will be renamed to simply Taxon in next major release
```
Comment (emodric) [inline lib/Layout/Resolver/TargetType/SingleTaxon.php]: Please use concrete version: `in 2.0`

File: lib/Layout/Resolver/TargetType/Taxon.php [hunk 7]
```diff
@@ -10,6 +10,9 @@
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Validator\Constraints;
 
+/**
+ * @deprecated this class will be renamed to TaxonTree in next major release
```
Comment (emodric) [inline lib/Layout/Resolver/TargetType/Taxon.php]: Please use concrete version: `in 2.0`

File: lib/Layout/Resolver/TargetHandler/Doctrine/SingleTaxon.php [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+ */
+final class SingleTaxon implements TargetHandlerInterface
+{
+    public function handleQuery(QueryBuilder $query, mixed $value): void
+    {
+        $query->andWhere(
+            $query->expr()->in('rt.value', [':target_value']),
```
Comment (emodric) [inline lib/Layout/Resolver/TargetHandler/Doctrine/SingleTaxon.php]: If this is a single taxon identifier, shouldn't this be `$query->expr()->eq('rt.value', ':target_value')` ?

File: lib/Layout/Resolver/TargetType/SingleTaxon.php [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        return [
+            new Constraints\NotBlank(),
+            new Constraints\Type(['type' => 'numeric']),
+            new Constraints\GreaterThan(['value' => 0]),
+            new SyliusConstraints\Taxon(),
+        ];
+    }
+
+    /**
+     * @return int[]|null
```
Comment (emodric) [inline lib/Layout/Resolver/TargetType/SingleTaxon.php]: This PHPDoc is wrong and superfluous :)

File: bundle/Resources/config/services/layout_resolver.yaml (hunk 1 above)
Comment (RandyCupic) [reply bundle/Resources/config/services/layout_resolver.yaml:32]: @emodric Fixed in https://github.com/netgen-layouts/layouts-sylius/pull/17/commits/057d41669cd3128b87eaf8063b07be5151bc5382

File: lib/Layout/Resolver/Form/TargetType/Mapper/SingleTaxon.php (hunk 2 above)
Comment (RandyCupic) [reply lib/Layout/Resolver/Form/TargetType/Mapper/SingleTaxon.php]: @emodric Fixed in https://github.com/netgen-layouts/layouts-sylius/pull/17/commits/057d41669cd3128b87eaf8063b07be5151bc5382

File: lib/Layout/Resolver/Form/TargetType/Mapper/Taxon.php (hunk 3 above)
Comment (RandyCupic) [reply lib/Layout/Resolver/Form/TargetType/Mapper/Taxon.php]: @emodric Fixed in https://github.com/netgen-layouts/layouts-sylius/pull/17/commits/057d41669cd3128b87eaf8063b07be5151bc5382

File: lib/Layout/Resolver/TargetHandler/Doctrine/SingleTaxon.php (hunk 4 above)
Comment (RandyCupic) [reply lib/Layout/Resolver/TargetHandler/Doctrine/SingleTaxon.php]: @emodric Fixed in https://github.com/netgen-layouts/layouts-sylius/pull/17/commits/057d41669cd3128b87eaf8063b07be5151bc5382

File: lib/Layout/Resolver/TargetHandler/Doctrine/Taxon.php (hunk 5 above)
Comment (RandyCupic) [reply lib/Layout/Resolver/TargetHandler/Doctrine/Taxon.php]: @emodric Fixed in https://github.com/netgen-layouts/layouts-sylius/pull/17/commits/057d41669cd3128b87eaf8063b07be5151bc5382

File: lib/Layout/Resolver/TargetType/SingleTaxon.php (hunk 6 above)
Comment (RandyCupic) [reply lib/Layout/Resolver/TargetType/SingleTaxon.php]: @emodric Fixed in https://github.com/netgen-layouts/layouts-sylius/pull/17/commits/057d41669cd3128b87eaf8063b07be5151bc5382

File: lib/Layout/Resolver/TargetType/Taxon.php (hunk 7 above)
Comment (RandyCupic) [reply lib/Layout/Resolver/TargetType/Taxon.php]: @emodric Fixed in https://github.com/netgen-layouts/layouts-sylius/pull/17/commits/057d41669cd3128b87eaf8063b07be5151bc5382

File: lib/Layout/Resolver/TargetHandler/Doctrine/SingleTaxon.php (hunk 8 above)
Comment (RandyCupic) [reply lib/Layout/Resolver/TargetHandler/Doctrine/SingleTaxon.php]: @emodric Fixed in https://github.com/netgen-layouts/layouts-sylius/pull/17/commits/057d41669cd3128b87eaf8063b07be5151bc5382

File: lib/Layout/Resolver/TargetType/SingleTaxon.php (hunk 9 above)
Comment (RandyCupic) [reply lib/Layout/Resolver/TargetType/SingleTaxon.php]: @emodric Fixed in https://github.com/netgen-layouts/layouts-sylius/pull/17/commits/057d41669cd3128b87eaf8063b07be5151bc5382

**Comment by emodric** [comment]
Thanks @RandyCupic !

**Pull request merged**